### ISSUE TITLE
learn: how to apply dispatch event

### ISSUE THREAD

**Issue by ikbhal** [ikbhal/history#9]

**Comment by ikbhal** [comment]
<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>keydown demo</title>
  <style>
  fieldset {
    margin-bottom: 1em;
  }
  input {
    display: block;
    margin-bottom: .25em;
  }
  #print-output {
    width: 100%;
  }
  .print-output-line {
    white-space: pre;
    padding: 5px;
    font-family: monaco, monospace;
    font-size: .7em;
  }
  </style>
  <script src="https://code.jquery.com/jquery-1.10.2.js"></script>
</head>
<body>
 
<form>
  <fieldset>
    <label for="target">Type Something:</label>
    <input id="target" type="text">
  </fieldset>
</form>
<button id="other">
  Trigger the handler
</button>

 
<script>
var xTriggered = 0;
$( "#target" ).keydown(function( event ) {
  if ( event.which == 13 ) {
   event.preventDefault();
  }
  xTriggered++;
  var msg = "Handler for .keydown() called " + xTriggered + " time(s).";
  console.log(msg);
  console.log(event);
});
 
$( "#other" ).click(function() {
  $( "#target" ).keydown();
});
</script>
 
</body>
</html>

https://api.jquery.com/keydown/

**Comment by ikbhal** [comment]
https://jsfiddle.net/rickj33/3fby0nzr/
<p><label>input: <input type="text" name="foo" id="input" value="" /></label></p>

<p>
    <button class="keydown" data-code="13">Trigger keydown Enter</button>
    <button class="keypress" data-code="13">Trigger keypress Enter</button>
    <button class="keyup" data-code="13">Trigger keyup Enter</button>
</p>

<p>
    <button class="keydown" data-code="65">Trigger keydown 'a'</button>
    <button class="keypress" data-code="65">Trigger keypress 'a'</button>
    <button class="keyup" data-code="65">Trigger keyup 'a'</button>
</p>

<hr />

<div id="console"></div>
====
console.d = function(log){
    $('#console').text(log);
}

var callback = function(e){
    console.log(e.type, e);
    var text = e.type;
    var code = e.which ? e.which : e.keyCode;
    if(13 === code){
        text += ': ENTER';
    } else {
        text += ': keycode '+code;
    }
    console.d(text);
};

$('#input').keydown(callback);
$('#input').keypress(callback);
$('#input').keyup(callback);

$('.keydown').click(function(e){
    var code = $(this).data('code');
    $('#input').trigger(
        jQuery.Event( 'keydown', { keyCode: code, which: code } )
    );
});
$('.keypress').click(function(e){
    var code = $(this).data('code');
    $('#input').trigger(
        jQuery.Event( 'keypress', { keyCode: code, which: code } )
    );
});
$('.keyup').click(function(e){
    var code = $(this).data('code');
    $('#input').trigger(
        jQuery.Event( 'keyup', { keyCode: code, which: code } )
    );
});
---but still not working

**Comment by ikbhal** [comment]
not working above

**Comment by ikbhal** [comment]
https://stackoverflow.com/questions/596481/is-it-possible-to-simulate-key-press-events-programmatically/34095590

**Comment by ikbhal** [comment]
var keyboardEvent = document.createEvent("KeyboardEvent");
var initMethod = typeof keyboardEvent.initKeyboardEvent !== 'undefined' ? "initKeyboardEvent" : "initKeyEvent";

keyboardEvent[initMethod](
  "keydown", // event type: keydown, keyup, keypress
  true,      // bubbles
  true,      // cancelable
  window,    // view: should be window
  false,     // ctrlKey
  false,     // altKey
  false,     // shiftKey
  false,     // metaKey
  40,        // keyCode: unsigned long - the virtual key code, else 0
  0          // charCode: unsigned long - the Unicode character associated with the depressed key, else 0
);
document.dispatchEvent(keyboardEvent);

**Comment by ikbhal** [comment]
https://stackoverflow.com/questions/3368578/trigger-a-keypress-keydown-keyup-event-in-js-jquery

**Comment by ikbhal** [comment]
<html>
<head>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
<script>
$(document).ready(function(){
  $("#btn").click(function(){
    //alert("The paragraph was clicked.");
    console.log("btn click");
    var e = $.Event('keypress');
    e.which = 65; // Character 'A'
    console.log(e);
    $('#txt').trigger(e);
    $('#txt').on('keypress', function(e){
    	console.log("inside keydown");
    	console.log(e);
    });
  });
});
</script>
</head>
<body>
	<h1>dispatche vent</h1>
	<input type="text" id="txt"></input>
	<hr>
	<button id="btn">dispatch event</button>
</body>
</html>

not working
